development/editor/creating_icons.rst

```diff
@@ -0,0 +1,83 @@
+.. _doc_editor_icons:
+
+Editor icons
+============
+
+When a new class is created and exposed to scripting, the editor's interface
+will display it with a default icon representing the base class it's inherited
+from. Yet in most cases it is recommended to create icons for new classes
+to improve the user experience considerably.
+
+Creating icons
+~~~~~~~~~~~~~~
+
+In order to create new icons, you first need a vector graphics editor installed.
+For instance, you can use open-source `Inkscape <https://inkscape.org/>`_ editor.
+
+Clone the ``godot-design`` repository containing all the original editor icons:
+
+   .. code:: bash
+
+       git clone https://github.com/godotengine/godot-design
+
+The icons must be created in a vector graphics editor in ``svg`` format. You
+can use ``/engine/icons/inkscape_template.svg`` with default icon properties
+already set up.
+
+Once you're satisfied with the icon's design, save the icon in
+``engine/icons/svg/`` folder. But in order for the engine to automatically
+pick up the icons, each icon's filename:
+
+1. Must be prefixed with ``icon_``.
+
+2. Each consecutive capitalized letter in the class name must be converted to
```

Comment:
That's not really true, as this would lead to `iconc_s_g_p_olygon`. It's correct if you say "prefix" instead of "suffix", but it's still hard to understand IMO.

It would be better to say that the "PascalName" class name should be converted to "snake_case", so words are separated by `_` instead of case changes, and uppercase acronyms also have all letters separated as distinct words. You can give two polygons (one with an acronym and one without -- but maybe with numbers) to make it clearer.

Reply:
It's interesting how some icons are registered just fine for digits with and without `_`:
`icon_area_2d.svg`
`icon_bone_2_d.svg`

I guess people just use `PascalCase` → `snake_case` converter which leads to this inconsistency.
The documentation should accept the style dictated by these converters then.